src/network.zig

```diff
@@ -1048,78 +1049,66 @@ pub const Protocols = struct {
 								curTime = actualTime;
 							}
 						}
-						const newBiome = main.server.terrain.biomes.getById(zon.get([]const u8, "biome", ""));
+
+						const newBiome = main.server.terrain.biomes.getById(biomeId);
 						const oldBiome = world.playerBiome.swap(newBiome, .monotonic);
 						if(oldBiome != newBiome) {
 							main.audio.setMusic(newBiome.preferredMusic);
 						}
 					}
 				},
-				else => |unrecognizedType| {
-					std.log.err("Unrecognized type for genericUpdateProtocol: {}. Data: {any}", .{unrecognizedType, reader.remaining});
-					return error.Invalid;
-				},
 			}
 		}
 
-		fn addHeaderAndSendImportant(conn: *Connection, header: u8, data: []const u8) void {
-			const headeredData = main.stackAllocator.alloc(u8, data.len + 1);
-			defer main.stackAllocator.free(headeredData);
-			headeredData[0] = header;
-			@memcpy(headeredData[1..], data);
-			conn.sendImportant(id, headeredData);
-		}
-
-		fn addHeaderAndSendUnimportant(conn: *Connection, header: u8, data: []const u8) void {
-			const headeredData = main.stackAllocator.alloc(u8, data.len + 1);
-			defer main.stackAllocator.free(headeredData);
-			headeredData[0] = header;
-			@memcpy(headeredData[1..], data);
-			conn.sendUnimportant(id, headeredData);
-		}
-
 		pub fn sendGamemode(conn: *Connection, gamemode: main.game.Gamemode) void {
-			conn.sendImportant(id, &.{type_gamemode, @intFromEnum(gamemode)});
+			conn.sendImportant(id, &.{@intFromEnum(UpdateType.gamemode), @intFromEnum(gamemode)});
 		}
 
 		pub fn sendTPCoordinates(conn: *Connection, pos: Vec3d) void {
 			var writer = utils.BinaryWriter.initCapacity(main.stackAllocator, networkEndian, 25);
 			defer writer.deinit();
-			writer.writeInt(u8, type_teleport);
-			writer.writeInt(u64, @bitCast(pos[0]));
-			writer.writeInt(u64, @bitCast(pos[1]));
-			writer.writeInt(u64, @bitCast(pos[2]));
+
+			writer.writeEnum(UpdateType, .teleport);
+			writer.writeVec(Vec3d, pos);
+
 			conn.sendImportant(id, writer.data.items);
 		}
 
 		pub fn sendCure(conn: *Connection) void {
-			var data: [1]u8 = undefined;
-			data[0] = type_cure;
-			conn.sendImportant(id, &data);
+			var writer = utils.BinaryWriter.initCapacity(main.stackAllocator, networkEndian, 1);
+			defer writer.deinit();
+
+			writer.writeEnum(UpdateType, .cure);
+
+			conn.sendImportant(id, writer.data.items);
```

Comment:
If it's just one item, why not just use
```
conn.sendImportant(id, &.{@intFromEnum(UpdateType.cure)});
```
like above for the gamemode.

Reply:
Actually, this is just a relic from the java version which had a cure command. You can just remove this. It cannot work like this with the current synchronization approach anyways.

Reply:
Do we still care about interopt with java and want to keep those reserved values around?

Reply:
No, I guess they can be removed. it was intended for interop with older versions, but that never really worked out anyways.